lib/src/expect/throws_matcher.dart

```diff
@@ -73,7 +73,7 @@ class Throws extends AsyncMatcher {
   // function.
   @override
   dynamic /*FutureOr<String>*/ matchAsync(Object? item) {
-    if (item is! Function && item is! Future) {
+    if (item is! Function() && item is! Future) {
```

Comment:
This stopped accepting values which are a `Function<T>()`. I think we may need to keep this as a dynamic call to handle generic methods. This is why the `Future.value` tear-off case is failing.

cc @jakemac53 - I have completely overlooked the fact that functions are not subtypes if they have different numbers of generic type arguments... I wonder if there are any other places we have been more strict than we should and are unintentionally blocking generic functions used as callbacks.